### DISCUSSION TITLE
Accessing initial values of variable

### DISCUSSION THREAD

**Discussion by adigc** [idaholab/moose#25365] [Q&A General] [answered]
For a class I am implementing, when calcuating the residual,  I need the initial values of a variable. 
Is there a function similar to` _var.dofValues()` with which I can access the initial values of a DOF ? 
If not how can I get the intial values set from the input file. 

NOTE: I am setting inital value to the variable in the input file.

**Comment by GiudGiud** [comment]
Hello

So you want the initial value later on, as in you cannot rely on the current value of the variable?

Guillaume

**Reply by GiudGiud** [reply]
I would just create an auxiliary variable with the same initial value and not updating it

**Reply by adigc** [reply]
sounds good ... I will follow this approach. But then my question is, this approach will consume more memory no ?

**Reply by GiudGiud** [reply, marked as answer]
Sure. But it s not that much. Any time you add a new field you want to look at with paraview, you also incur that kind of cost. 
If you change the time integration scheme it ll store even more fields, there's plenty of ways to add an extra field that will cost you memory
